cryptographic_estimators/UOVEstimator/UOVAlgorithms/collision_attack.py

```diff
@@ -17,34 +17,28 @@
 
 from ..uov_algorithm import UOVAlgorithm
 from ..uov_problem import UOVProblem
-from ...MQEstimator.mq_estimator import MQEstimator
 from ...base_algorithm import optimal_parameter
-from ...helper import round_or_truncate
-from ...base_constants import BASE_MEMORY_BOUND, BASE_NSOLUTIONS, BASE_BIT_COMPLEXITIES, BASE_EXCLUDED_ALGORITHMS
 from math import log2, e
 from cryptographic_estimators.base_constants import BASE_FORGERY_ATTACK
 
 
 
 class CollisionAttack(UOVAlgorithm):
-    """
-    Construct an instance of CollisionAttack estimator
-
-    Collision attack is a general attack which works against any signature which 
-    follows the hash-and-sign paradigm. 
-
-    INPUT:
-
-    - ``problem`` -- an instance of the UOVProblem class
-    - ``gray_code_eval_cost`` -- logarithm of the cost to evaluate one polynomial in one vector using Gray-code enumeration (default: log(q))
-    - ``X`` -- Number of preimages
-    - ``Y`` -- Number of variables in the salt space
-    - ``memory_access`` -- specifies the memory access cost model (default: 0, choices: 0 - constant, 1 - logarithmic, 2 - square-root, 3 - cube-root or deploy custom function which takes as input the logarithm of the total memory usage)
-    - ``complexity_type`` -- complexity type to consider (0: estimate, 1: tilde O complexity, default: 0)
-
-    """
-
     def __init__(self, problem: UOVProblem, **kwargs):
+        """Construct an instance of CollisionAttack estimator.
+
+        Collision attack is a general attack which works against any signature which 
+        follows the hash-and-sign paradigm. 
+
+        Args:
+            problem (UOVProblem): An instance of the UOVProblem class
+            **kwargs: Additional keyword arguments
+                gray_code_eval_cost (float): Logarithm of the cost to evaluate one polynomial in one vector using Gray-code enumeration (default: log(q))
+                X (int): Number of preimages
+                Y (int): Number of variables in the salt space
+                memory_access (int): Specifies the memory access cost model (default: 0, choices: 0 - constant, 1 - logarithmic, 2 - square-root, 3 - cube-root or deploy custom function which takes as input the logarithm of the total memory usage)
```

Comment:
wrong indentation

Reply:
to be fixed in  #198

Reply:
Nothing to fix. Indentation is correctly parsed by our custom Sphinx extension (see https://github.com/Crypto-TII/CryptographicEstimators/pull/211).